pkg/apis/pipeline/v1/container_types.go

```diff
@@ -142,6 +142,10 @@ type Step struct {
 	// +optional
 	// +listType=atomic
 	Params Params `json:"params,omitempty"`
+	// Results declares step results produced by the step.
+	// +optional
+	// +listType=atomic
+	Results []StepResult `json:"results,omitempty"`
```

Comment:
Could you remind me of how are we going to use this `Results`?  May also be helpful for other reviewers. 
And are we going to have validations between Step.Results and StepAction.Results?

Reply:
The use is to inform the entrypointer to extract these step results from another location `/tekton/steps/stepName/results/resultName`.